ttkwidgets/tickscale.py

```diff
@@ -0,0 +1,633 @@
+        if self._showvalue:
+            self.label.configure(text=self._formatter.format(float(value)))
+            self.update_idletasks()
+            if self._labelpos in ['n', 's']:
+                # position (in pixel) of the center of the slider
+                x = self.convert_to_pixels(float(value))
+                # pay attention to the borders
+                half_width = self.label.winfo_width() / 2
+                if x + half_width > self.scale.winfo_width():
+                    x = self.scale.winfo_width() - half_width
+                elif x - half_width < 0:
+                    x = half_width
+                self.label.place_configure(x=x)
+
+    def _display_value_vertical(self, value):
+        """
+        Display the current value and update the label position.
+        Return the pady necessary to display the label.
+        """
+        if self._showvalue:
+            self.label.configure(text=self._formatter.format(float(value)))
+            if self._labelpos in ['e', 'w']:
+                y = self.convert_to_pixels(float(value))
+                self.label.place_configure(y=y)
+
+    def _place_ticks_horizontal(self):
+        # first tick
+        tick = self.ticks[0]
+        label = self.ticklabels[0]
+        x = self.convert_to_pixels(tick)
+        half_width = label.winfo_reqwidth() / 2
+        if x - half_width < 0:
+            x = half_width
+        label.place_configure(x=x)
+        # ticks in the middle
+        for tick, label in zip(self.ticks[1:-1], self.ticklabels[1:-1]):
+            x = self.convert_to_pixels(tick)
+            label.place_configure(x=x)
+        # last tick
+        tick = self.ticks[-1]
+        label = self.ticklabels[-1]
+        x = self.convert_to_pixels(tick)
+        half_width = label.winfo_reqwidth() / 2
+        if x + half_width > self.scale.winfo_reqwidth():
+            x = self.scale.winfo_width() - half_width
+        label.place_configure(x=x)
+
+    def _place_ticks_vertical(self):
+        for tick, label in zip(self.ticks, self.ticklabels):
+            y = self.convert_to_pixels(tick)
+            label.place_configure(y=y)
+
+    def _increment(self, *args):
+        """Move the slider only by increment given by resolution."""
+        value = self._var.get()
+        if self._resolution:
+            value = self._start + int(round((value - self._start) / self._resolution)) * self._resolution
+            self._var.set(value)
+        self.display_value(value)
+
+    def _style_change(self, event=None):
+        """Apply style and update widgets position."""
+        self._apply_style()
+        self._init()
+
+    def _update_display(self, event=None):
+        """Redisplay the ticks and the label so that they adapt to the new size of the scale."""
+        try:
+            if self._showvalue:
+                self.display_value(self.scale.get())
+            if self._tickinterval:
+                self.place_ticks()
+        except Exception:
```

Comment:
Could this except clause be made to catch a `TclError` instead?

Reply:
It now catches the actual error: `IndexError`.